x/gamm/keeper/swap.go

```diff
@@ -14,21 +14,22 @@ import (
 // denominated via tokenOutDenom through a pool denoted by poolId specifying that
 // tokenOutMinAmount must be returned in the resulting asset returning an error
 // upon failure. Upon success, the resulting tokens swapped for are returned. A
-// swap fee is applied determined by the pool's parameters.
+// swap fee is applied determined by the pool's parameters and given multiplier.
 func (k Keeper) SwapExactAmountIn(
 	ctx sdk.Context,
 	sender sdk.AccAddress,
 	poolId uint64,
 	tokenIn sdk.Coin,
 	tokenOutDenom string,
 	tokenOutMinAmount sdk.Int,
+	swapFeeMultiplier sdk.Dec,
 ) (sdk.Int, error) {
 	pool, err := k.getPoolForSwap(ctx, poolId)
 	if err != nil {
 		return sdk.Int{}, err
 	}
 
-	swapFee := pool.GetSwapFee(ctx)
+	swapFee := pool.GetSwapFee(ctx).Mul(swapFeeMultiplier)
 	return k.swapExactAmountIn(ctx, sender, pool, tokenIn, tokenOutDenom, tokenOutMinAmount, swapFee)
```

Comment:
Is there a specific reason you decided to have the swapFeeMultiplier as an input to SwapExactAmountIn/Out? It seems like having the logic directly in the SwapExactAmountIn/Out function itself would be potentially safer/cleaner (I could also just totally be missing something!). Lmk what you think

Reply:
This functions performs only single swaps. That mean, them don't know context executions - whether it is multihop swaps or just single swap. But i feel something as you feel - that it is a bit strange to have that multiplier there. Wii try to create another version where multihop logic will use private keeper functions.